runtime/compiler/ilgen/IlGenerator.cpp

```diff
@@ -558,13 +558,24 @@ TR_J9ByteCodeIlGenerator::genILFromByteCodes()
 
       if (currNode->getOpCodeValue() == TR::checkcast
           && currNode->getSecondChild()->getOpCodeValue() == TR::loadaddr
-          && currNode->getSecondChild()->getSymbolReference()->isUnresolved()
-          && // check whether the checkcast class is primitive valuetype. Expansion is only needed for checkcast to reference type.
-            (!TR::Compiler->om.areValueTypesEnabled()
-            || !TR::Compiler->cls.isClassRefPrimitiveValueType(comp(), method()->classOfMethod(),
-                                        currNode->getSecondChild()->getSymbolReference()->getCPIndex())))
+          && currNode->getSecondChild()->getSymbolReference()->isUnresolved())
           {
-          unresolvedCheckcastTopsNeedingNullGuard.add(currTree);
+          if (TR::Compiler->om.areValueTypesEnabled())
+            {
+            if (TR::Compiler->om.doNotSupportQDescriptorForValueTypes())
+               {
+               TR_ASSERT_FATAL(false, "TO BE IMPLEMENTED!!!");
```

Comment:
I feel it'd be better to keep it for now in case of the spec changes.  We will reevaluate it when we implement the case where Q signature is not supported (`!TR::Compiler->om.isQDescriptorForValueTypesSupported()`).

Reply:
After reading the latest spec change, I think it's safe to remove the special handling on null check in checkcast for value types. Also disabled `testCheckCastValueTypeOnNull`. Addressed in c4346cc